### PR TITLE
[feat] add permission and role model/routes/tests

### PR DESCRIPTION
**Pull request by thmarinho**

### CODE REVIEW

File: server/routes/tag.js [hunk 1]
```diff
@@ -1,7 +1,12 @@
 const router = require("express").Router();
 const tagModel = require("../models/Tag");
+const {checkPermission} = require("../middleware/permission");
+const { checkTokenValidity } = require("../middleware/token");
 
-router.get("/", async (req, res) => {
+router.use(checkTokenValidity);
+router.use(checkPermission);
+
+router.get("/", checkPermission, async (req, res) => {
```
Comment (thmarinho) [inline server/routes/tag.js:9]: will be fix

File: server/sse/index.js [hunk 2]
```diff
@@ -1,9 +1,9 @@
 const router = require("express").Router();
 const broker = require("../messages/index");
 const banModel = require("../models/Ban");
-const {checkAuthentication} = require("../middleware");
+const checkTokenValidity = require("../middleware/token").checkTokenValidity;
```
Comment (thmarinho) [inline server/sse/index.js:4]: will be fix

Comment (RemyBvd) [inline server/sse/index.js:4]: Changer l'export pour récupérer directement le token

**Comment by thmarinho** [comment]
> you should probably add a check for an admin role to allow access to all routes

Ant idea to do this correctly? @matfire

**Pull request merged**